ql/src/test/queries/clientpositive/cbo_rp_null_order.q

```diff
@@ -0,0 +1,7 @@
+SET hive.cbo.returnpath.hiveop=true;
+SET hive.default.nulls.last=false;
+
+CREATE TABLE t1(key int, value string);
+
+EXPLAIN CBO SELECT * FROM t1 a INNER JOIN t1 b on a.key = b.key;
```

Comment:
I think that for INNER JOIN the direction of nulls is not gonna affect correctness. Does the null direction affect correctness of outer joins?

Reply:
I think it has an affect only on queries with limit but without order/sort by. However it doesn't matter which n rows do we pick.
We can see the null sort order in the RS properties.
```
                      Reduce Output Operator
                        key expressions: key (type: string)
                        null sort order: z
                        sort order: +
```
The join is only necessary to get the rule `HiveInsertExchange4JoinRule` to add the `HiveSortExchange` operators.